src/braket/analog_hamiltonian_simulator/rydberg/rydberg_simulator_helpers.py

```diff
@@ -0,0 +1,432 @@
+        detuning_coef = np.array(
+            [
+                _interpolate_time_series(
+                    t, detuning.times, detuning.values, method="piecewise_linear"
+                )
+                for t in simulation_times
+            ],
+            dtype=complex,
+        )
+        detuning_coefs.append(detuning_coef)
+
+    # add shifting fields
+    local_detuing_coefs = []
+    for shifting_field in program.hamiltonian.shiftingFields:
+        magnitude = shifting_field.magnitude.time_series
+
+        local_detuing_coef = np.array(
+            [
+                _interpolate_time_series(
+                    t, magnitude.times, magnitude.values, method="piecewise_linear"
+                )
+                for t in simulation_times
+            ],
+            dtype=complex,
+        )
+        local_detuing_coefs.append(local_detuing_coef)
+
+    return np.array(rabi_coefs), np.array(detuning_coefs), np.array(local_detuing_coefs)
+
+
+def get_ops_coefs(
+    program: Program,
+    configurations: List[str],
+    rydberg_interaction_coef: float,
+    simulation_times: List[float],
+) -> Tuple[
+    List[scipy.sparse.csr_matrix],
+    List[scipy.sparse.csr_matrix],
+    List[scipy.sparse.csr_matrix],
+    np.ndarray,
+    np.ndarray,
+    np.ndarray,
+    scipy.sparse.csr_matrix,
+]:
+    """Returns the sparse matrices and coefficients for the Rabi terms, detuning terms and
+    the local detuining terms, together with the interaction operator in the given analog
+    simulation program for Rydberg systems.
+
+    Args:
+        program (Program): An analog simulation program for Rydberg system
+        configurations (List[str]): The list of configurations that comply to the
+            blockade approximation.
+        rydberg_interaction_coef (float): The interaction coefficient
+        simulation_times (List[float]): The list of time points
+
+    Returns:
+        Tuple[
+            List[csr_matrix],
+            List[csr_matrix],
+            List[csr_matrix],
+            ndarray,
+            ndarray,
+            ndarray,
+            csr_matrix
+        ]: A tuple containing the list of Rabi operators, the list of detuing operators,
+        the list of local detuing operators, the list of Rabi frequencies, the list of global
+        detuings, the list of local detunings and the interaction operator.
+
+    """
+
+    rabi_ops, detuning_ops, interaction_op, local_detuning_ops = _get_sparse_ops(
+        program, configurations, rydberg_interaction_coef
+    )
+    rabi_coefs, detuning_coefs, local_detuing_coefs = _get_coefs(program, simulation_times)
+
+    return (
+        rabi_ops,
+        detuning_ops,
+        local_detuning_ops,
+        rabi_coefs,
+        detuning_coefs,
+        local_detuing_coefs,
+        interaction_op,
+    )
+
+
+def sample_state(state: np.ndarray, shots: int) -> np.ndarray:
+    """Sample measurement outcomes from the quantum state `stat`
+    Args:
```

Comment:
nit: spacing and spelling
```suggestion
    """Sample measurement outcomes from the quantum state `state`
    
    Args:
```